XenAdmin/TabPages/ManageUpdatesPage.cs

```diff
@@ -0,0 +1,1418 @@
+                }
+            }
+        }
+
+        private void ToolStripMenuItemDownloadNewXenCenter_Click(object sender, EventArgs e)
+        {
+            DataGridViewRow clickedRow = FindAlertRow(sender as ToolStripMenuItem);
+            if (clickedRow == null)
+                return;
+
+            XenServerUpdateAlert updateAlert = (XenServerUpdateAlert)clickedRow.Tag;
+
+            if (updateAlert == null || updateAlert.RequiredClientVersion == null)
+                return;
+
+            string xenCenterUrl = updateAlert.RequiredClientVersion.Url;
+            if (string.IsNullOrEmpty(xenCenterUrl))
+                return;
+
+            Program.Invoke(Program.MainWindow, () => Program.OpenURL(xenCenterUrl));
+        }
+
+        private void ToolStripMenuItemCopy_Click(object sender, EventArgs e)
+        {
+            DataGridViewRow clickedRow = FindAlertRow(sender as ToolStripMenuItem);
+            if (clickedRow == null)
+                return;
+
+            StringBuilder sb = new StringBuilder();
+            
+            if (dataGridViewUpdates.SelectedRows.Count > 1 && dataGridViewUpdates.SelectedRows.Contains(clickedRow))
+            {
+                foreach (DataGridViewRow r in dataGridViewUpdates.SelectedRows)
+                {
+                    Alert alert = (Alert) r.Tag;
+                    sb.AppendLine(alert.GetUpdateDetailsCSVQuotes());
+                }
+            }
+            else
+            {
+                Alert alert = (Alert) clickedRow.Tag;
+                if (alert != null) 
+                    sb.AppendLine(alert.GetUpdateDetailsCSVQuotes());
+            }
+
+            Clip.SetClipboardText(sb.ToString());
+        }
+
+        #endregion
+
+        #region DataGridView event handlers
+
+        private void dataGridViewUpdates_SelectionChanged(object sender, EventArgs e)
+        {
+            if (_buildInProgress)
+                return;
+
+            UpdateButtonEnablement();
+        }
+
+        private void dataGridViewUpdates_ColumnHeaderMouseClick(object sender, DataGridViewCellMouseEventArgs e)
+        {
+            if (dataGridViewUpdates.Columns[e.ColumnIndex].SortMode == DataGridViewColumnSortMode.Automatic)
+                Rebuild();
+        }
+
+        /// <summary>
+        /// Handles the automatic sorting of the AlertsGridView for the non-string columns
+        /// </summary>
+        private void dataGridViewUpdates_SortCompare(object sender, DataGridViewSortCompareEventArgs e)
+        {
+            Alert alert1 = (Alert)dataGridViewUpdates.Rows[e.RowIndex1].Tag;
+            Alert alert2 = (Alert)dataGridViewUpdates.Rows[e.RowIndex2].Tag;
+
+            if (e.Column.Index == ColumnDate.Index)
+            {
+                int sortResult = DateTime.Compare(alert1.Timestamp, alert2.Timestamp);
+                e.SortResult = (dataGridViewUpdates.SortOrder == SortOrder.Descending) ? sortResult *= -1 : sortResult;

```

Comment:
Apart from this, sorting was not working correctly. I have fixed both issues in further commits related to the CDN updates, so it will be in a next PR.